Helper/Order.php

```diff
@@ -1336,8 +1336,19 @@ public function setOrderState($order, $state)
                 $order->setStatus($order->getConfig()->getStateDefaultStatus(OrderModel::STATE_HOLDED));
             }
         } elseif ($state == OrderModel::STATE_CANCELED) {
-            $this->cancelOrder($order);
-            return;
+            if ($order->canCancel()){
+                $this->cancelOrder($order);
+                return;
+            }
+
+            try{
+                // Restock product quantity when the payment is irreversibly rejected
+                $order->registerCancellation('', false);
+            } catch (\Exception $e){
+                // Put the order in "cancelled" state even if the previous call fails
```

Comment:
Let's bump a metric here stating that we fell back.

I'm curious as to why this happens in the first place

Reply:
@roopakv, I didn't see any exceptions thrown while doing the development work. This is just in case an exception occurs and I just followed the same pattern that existed before. See here https://github.com/BoltApp/bolt-magento2/blob/master/Helper/Order.php#L1335

Let me know if you think I should remove the try-catch block.

Reply:
Let's keep it but bump a metric.this Isc The perfect case of something we don't expect to happen happening. We can then track if this happens and how many times

Reply:
@roopakv can you explain what you mean by "bump a metric"? Are you saying that I should add a Bugsnag notification, or bump the Bolt version number, or something else?